prometheus/promhttp/instrument_server.go

```diff
@@ -132,12 +129,12 @@ func InstrumentHandlerTimeToWriteHeader(obs prometheus.ObserverVec, next http.Ha
 
 	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
 		now := time.Now()
-		d := &wroteHeaderDelegator{
-			delegator: newDelegator(w),
+		d := newDelegator(&responseWriterDelegator{
```

Comment:
This will again kill the interface upgrade. I think the way to go is to change `newDelegator` to also accept the observeWriteHeader func (which can be `nil`) and set it in the created `responseWriterDelegator`. It doesn't look beautiful, but at least `newDelegator` is not exported.